pkg/channel/fanout/fanout_message_handler.go

```diff
@@ -203,16 +245,14 @@ func parseFanoutResultAndReportMetrics(result dispatchResult, reporter channel.S
 	return err
 }
 
-// dispatch takes the event, fans it out to each subscription in f.config. If all the fanned out
+// dispatch takes the event, fans it out to each subscription in subs. If all the fanned out
 // events return successfully, then return nil. Else, return an error.
-func (f *MessageHandler) dispatch(ctx context.Context, bufferedMessage binding.Message, additionalHeaders nethttp.Header) dispatchResult {
-	subs := len(f.config.Subscriptions)
-
+func (f *FanoutMessageHandler) dispatch(ctx context.Context, subs []Subscription, bufferedMessage binding.Message, additionalHeaders nethttp.Header) dispatchResult {
 	// Bind the lifecycle of the buffered message to the number of subs
-	bufferedMessage = buffering.WithAcksBeforeFinish(bufferedMessage, subs)
+	bufferedMessage = buffering.WithAcksBeforeFinish(bufferedMessage, len(subs))
 
-	errorCh := make(chan dispatchResult, subs)
-	for _, sub := range f.config.Subscriptions {
+	errorCh := make(chan dispatchResult, len(subs))
```

Comment:
`errorCh` reads to me that there should only be errors on the chan, but later down you assume there will be an entry on this channel for every sub. So maybe more like `resultCh`